store/src/java/com/zimbra/cs/service/account/EndSession.java

```diff
@@ -56,13 +56,13 @@ public Element handle(Element request, Map<String, Object> context)
             try {
                 AuthToken at = zsc.getAuthToken();
                 if (at.isJWT()) {
```

Comment:
I would rather encapsulate the logic that removes token from session in the class that implements the token instead of having if/else branches inside EndSession.java.

Reply:
Inside the auth token class, there is no reference of the context which holds the http req and response, hence it won't be possible to set the cookie inside the auth token class. Thats why this logic is here in EndSession.